From 34af6ba1a86abc20b32c49136fc2bf1ccd0a6355 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Wed, 16 Nov 2016 17:59:13 +0200
Subject: [PATCH 1099/2241] dts: a80x0: a70x0: add comphy handles to emac nodes

- add comphy handles to emac nodes
- change 10G port KR mode to SFI mode

Change-Id: Ic3bbe3752f51a277295e3582473ac46f0af7ac51
Reviewed-on: http://vgitil04.il.marvell.com:8080/33859
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts   |  4 +++-
 arch/arm64/boot/dts/marvell/armada-7040-db-router.dts |  4 ++++
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts      | 10 +++++++---
 arch/arm64/boot/dts/marvell/armada-8040-db-router.dts | 10 +++++++---
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts     | 12 ++++++++++--
 arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi  |  2 ++
 6 files changed, 33 insertions(+), 9 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
index ef818ed..ff2e823 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
@@ -126,7 +126,9 @@
 			};
 			gop {
 				emac0: mac0 {
-					phy-mode = "kr"; /* lane-2 */
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
 				};
 				emac2: mac2 {
 					phy-mode = "rgmii"; /* rgmii-0 */
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
index 095d5c6..a78f081 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
@@ -127,10 +127,14 @@
 			gop {
 				emac0: mac0 {
 					phy-mode = "sgmii"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SGMII0>;
+					phy-names = "comphy";
 				};
 				emac2: mac2 {
 					phy-mode = "sgmii"; /* lane-0 */
 					phy = <&phy2>;
+					phys = <&comphy0 0 COMPHY_SGMII1>;
+					phy-names = "comphy";
 				};
 				emac3: mac3 {
 					phy-mode = "rgmii"; /* rgmii-1 */
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index 42040d6..e84a9b1 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -80,10 +80,12 @@
 			};
 			gop {
 				emac0: mac0 {
-					phy-mode = "kr"; /* lane-2 */
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
 				};
 				emac2: mac2 {
-					phy-mode = "rgmii"; /* lane-0 */
+					phy-mode = "rgmii"; /* rgmii-0 */
 					phy = <&phy2>;
 				};
 				emac3: mac3 {
@@ -130,7 +132,9 @@
 			};
 			gop {
 				emac0_1: mac0 {
-					phy-mode = "kr"; /* lane-2 */
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy1 2 COMPHY_SFI>;
+					phy-names = "comphy";
 				};
 				emac2_1: mac2 {
 					phy-mode = "rgmii"; /* rgmii-0 */
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
index ffbefb4..b6e9113 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
@@ -82,10 +82,12 @@
 			};
 			gop {
 				emac0: mac0 {
-					phy-mode = "kr"; /* lane-2 */
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy0 2 COMPHY_SFI>;
+					phy-names = "comphy";
 				};
 				emac2: mac2 {
-					phy-mode = "rgmii"; /* lane-0 */
+					phy-mode = "rgmii"; /* rgmii-0 */
 					phy = <&phy2>;
 				};
 				emac3: mac3 {
@@ -136,7 +138,9 @@
 			};
 			gop {
 				emac0_1: mac0 {
-					phy-mode = "kr"; /* lane-2 */
+					phy-mode = "sfi"; /* lane-2 */
+					phys = <&comphy1 2 COMPHY_SFI>;
+					phy-names = "comphy";
 				};
 				emac2_1: mac2 {
 					phy-mode = "rgmii"; /* rgmii-0 */
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index b27f86e..b453c36 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -116,7 +116,9 @@
 			};
 			gop {
 				emac0_0: mac0 {
-					phy-mode = "kr"; /* lane-4 */
+					phy-mode = "sfi"; /* lane-4 */
+					phys = <&comphy0 4 COMPHY_SFI>;
+					phy-names = "comphy";
 				};
 			};
 			ppv22@000000 {
@@ -166,13 +168,19 @@
 			};
 			gop {
 				emac0_1: mac0 {
-					phy-mode = "kr"; /* lane-4 */
+					phy-mode = "sfi"; /* lane-4 */
+					phys = <&comphy1 4 COMPHY_SFI>;
+					phy-names = "comphy";
 				};
 				emac2_1: mac2 {
 					phy-mode = "sgmii"; /* lane-0 */
+					phys = <&comphy1 0 COMPHY_SGMII1>;
+					phy-names = "comphy";
 				};
 				emac3_1: mac3 {
 					phy-mode = "sgmii"; /* lane-5 */
+					phys = <&comphy1 5 COMPHY_SGMII2>;
+					phy-names = "comphy";
 				};
 			};
 			ppv22@000000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
index f2265d6..dc9f7c2 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-master.dtsi
@@ -17,6 +17,8 @@
  * Device Tree file for Marvell Armada CPN-110 Master.
  */
 
+#include <dt-bindings/phy/phy-mvebu-comphy.h>
+
 / {
 	cpn-110-master {
 		#address-cells = <2>;
-- 
2.7.4

